src/onehot.jl

```diff
@@ -18,6 +18,20 @@ function Base.:*(A::AbstractMatrix, b::OneHotVector)
   return A[:, b.ix]
 end
 
+function Base.:*(A::LinearAlgebra.Adjoint{T1,<:AbstractArray{T,2}}, b::OneHotVector) where {T1,T}
+  if size(A, 2) != b.of
+    throw(DimensionMismatch("Adjoint matrix column must correspond with OneHotVector size"))
+  end
+  return A[:, b.ix]
+end
+
+function Base.:*(A::LinearAlgebra.Adjoint{T1,<:AbstractArray{T,1}}, b::OneHotVector) where {T1<:Number,T}
+  if size(A, 2) != b.of
+    throw(DimensionMismatch("Adjoint matrix column must correspond with OneHotVector size"))
```

Comment:
Should this allow transpose too?
```suggestion
function Base.:*(A::AdjOrTransAbsVec, b::OneHotVector)
  if size(A, 2) != b.of
    throw(DimensionMismatch("length of row vector must match that of OneHotVector"))
```

Reply:
I think we'll need to have a separate methods for function `*`, for first argument as `Adjoint` or `Transpose` instead of single method as `Union{Adjoint, Transpose}`.

Ideally we would like to reduce such code-reduction, but I'm not sure if there any way to do that here?

Reply:
Oh it is possible those are separated in LinearAlgebra's definitions. As of 1.5 there might be restrictions `T<:Number` too. 

Perhaps both the matrix and row-vector definitions can be inside a `for Wrap in [:Adjoint, :Transpose] @eval begin ...` loop?